src/main/java/edu/cornell/kfs/module/cam/web/filter/CuCapAssetInventoryServerAuthFilter.java

```diff
@@ -34,30 +49,77 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
             response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
             response.getWriter().println(new Gson().toJson(CuCamsConstants.CapAssetApi.UNAUTHORIZED));
         }
     }
 
     private boolean isAuthorized(HttpServletRequest request) {
-        //todo switch to verify Cognito User Pool Key in request Header with Cognito public key
-        String correctApiKey = getWebServiceCredentialService().getWebServiceCredentialValue(CuCamsConstants.CapAssetApi.CAPITAL_ASSET_CREDENTIAL_GROUP_CODE,
-                CuCamsConstants.CapAssetApi.CAPITAL_ASSET_API_KEY_CREDENTIAL_NAME);
-        String submittedApiKey = request.getHeader(CuCamsConstants.CapAssetApi.CAPITAL_ASSET_API_KEY_CREDENTIAL_NAME);
-        return !StringUtils.isEmpty(correctApiKey) && !StringUtils.isEmpty(submittedApiKey) && submittedApiKey.equals(correctApiKey);
+        String cognitoIdToken = request.getHeader(CuCamsConstants.CapAssetApi.COGNITO_ID_TOKEN);
+        PublicKey cognitoUserPoolPublicKey = getPublicKeyFromFile();
+        if (ObjectUtils.isNull(cognitoUserPoolPublicKey)) {
+            return false;
+        }
+
+        Algorithm algorithm = Algorithm.RSA256((RSAPublicKey) cognitoUserPoolPublicKey, null);
+        JWTVerifier verifier = JWT.require(algorithm)
+                .withIssuer(CuCamsConstants.CapAssetApi.COGNITO_USER_POOL_ISSUER_URL)
+                .withClaim(CuCamsConstants.CapAssetApi.TOKEN_USE, CuCamsConstants.CapAssetApi.ID)
+                .build();
+
+        DecodedJWT jwt = verifier.verify(cognitoIdToken);
+        String email = jwt.getClaim(CuCamsConstants.CapAssetApi.EMAIL).asString();
+        LOG.info("CapAssetInventory Authorized {}", email);
+        return true;
+    }
+
+    private PublicKey getPublicKeyFromFile() {
+        try {
+            String cognitoPublicKeyPath = CuCamsConstants.CapAssetApi.COGNITO_PUBLIC_KEY_FILE_RELATIVE_PATH;
+            String cognitoPublicKeyFileContents = new String(Files.readAllBytes(Paths.get(cognitoPublicKeyPath)));
+            JsonObject publicKeyJson = parsePublicKeyFile(cognitoPublicKeyFileContents);
+            return decodePublicKey(publicKeyJson);
+        } catch (IOException | InvalidKeySpecException | NoSuchAlgorithmException ex) {
+            LOG.error("getPublicKeyFromFile", ex);
+        }
+        return null;
+    }
+
+    private JsonObject parsePublicKeyFile(String rawPublicKey) {
+        Gson gson = new Gson();
+        JsonElement jsonElement = gson.fromJson(rawPublicKey, JsonElement.class);
+        return jsonElement.getAsJsonObject();
+    }
+
+    private PublicKey decodePublicKey(JsonObject publicKeyJson) throws NoSuchAlgorithmException, InvalidKeySpecException {
+        String keyN = publicKeyJson.get("n").getAsString();
+        String keyE = publicKeyJson.get("e").getAsString();
+
+        Base64.Decoder decoder = keyN.endsWith("=") || keyN.contains("+") || keyN.contains("/") ? Base64.getDecoder() : Base64.getUrlDecoder();
```

Comment:
What is the significance of the "=", "+" and "/" characters in the "n" key?

Reply:
I have refactored this to always use the URL decoder.